service/src/main/java/com/andersen/service/auth/AuthServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+                throw new AuthenticationException("Invalid username or password");
+            }
+
+            this.currentUser = user;
+            OutputLogger.log(operation + " - Successful login for user: " + username);
+            return user;
+        } catch (DataAccessException e) {
+            String errorMsg = operation + " - Error during authentication";
+            ConsoleLogger.getLogger(AuthServiceImpl.class).error(errorMsg, e);
+            throw new AuthenticationException("Error during authentication"+e);
+        }
+    }
+
+    @Override
+    public User registerCustomer(String username, String password, String email, String fullName)
+            throws RegistrationException {
+        String operation = "User Registration";
+        OutputLogger.log(String.format("%s - Attempting to register new customer: %s (%s)",
+                operation, username, email));
+
+        try {
+            if (userRepository.getUserByUsername(username).isPresent()) {
+                String errorMsg = operation + " - Username already exists: " + username;
+                ConsoleLogger.getLogger(AuthServiceImpl.class).warn(errorMsg);
+                throw new RegistrationException("Username already exists");
+            }
+
+            User newUser = new User();
+            newUser.setUsername(username);
+            newUser.setPassword(hashPassword(password));
+            newUser.setEmail(email);
+            newUser.setFullName(fullName);
+            newUser.setRole(UserRole.CUSTOMER);
+
+            User createdUser = userRepository.createUser(newUser);
+            this.currentUser = createdUser;
+            OutputLogger.log(operation + " - Successfully registered new customer with ID: " + createdUser.getId());
+            return createdUser;
+        } catch (DataAccessException | AuthenticationException e) {
+            String errorMsg = operation + " - Error during registration";
+            ConsoleLogger.getLogger(AuthServiceImpl.class).error(errorMsg, e);
+            throw new RegistrationException("Error during registration", e);
+        }
+    }
+
+    @Override
+    public void logout() {
+        if (currentUser != null) {
+            OutputLogger.log("User Logout - Logging out user: " + currentUser.getUsername());
+            this.currentUser = null;
+        } else {
+            OutputLogger.log("User Logout - No user was logged in");
+        }
+    }
+
+    @Override
+    public User getCurrentUser() {
+        if (currentUser != null) {
+            ConsoleLogger.getLogger(AuthServiceImpl.class).debug(
+                    "Current user requested: " + currentUser.getUsername());
+        } else {
+            ConsoleLogger.getLogger(AuthServiceImpl.class).debug("No current user");
+        }
+        return currentUser;
+    }
+
+    @Override
+    public boolean isAdmin() {
+        boolean isAdmin = currentUser != null && currentUser.getRole() == UserRole.ADMIN;
```

Comment:
ConsoleLogger.getLogger(AuthServiceImpl.class) should be constant, should not be called in every method